pkg/pillar/containerd/oci.go

```diff
@@ -378,11 +381,22 @@ func (s *ociSpec) updateFromImageConfig(config v1.ImageConfig) error {
 	return oci.WithAdditionalGIDs("root")(ctrdCtx, s.client.ctrdClient, &dummy, &s.Spec)
 }
 
-// UpdateMounts adds volume specification mount points to the OCI runtime spec
+// UpdateMounts cleans old mounts and adds volume specification mount points to the OCI runtime spec
 func (s *ociSpec) UpdateMounts(disks []types.DiskStatus) error {
 	ociVolumeData := "rootfs"
 	blkMountPoints := ""
 
+	i := 0
+	for _, el := range s.Mounts {
+		//keep mounts with source to other location than vault or pointing to loader destination
+		if !strings.HasPrefix(el.Source, "/persist/vault") || el.Destination == loaderDestination {
+			s.Mounts[i] = el
+			i++
+		}
+	}
+	//truncate to actual size
+	s.Mounts = s.Mounts[:i]
```

Comment:
While looping over s.Mounts and updating s.Mounts work (since you keep a subset), it could result in odd bugs if this code is modified.
Thus it seems more safe to use a local variable mounts and append to that, and then assign s.Mounts = mounts at the end.

Probably makes sense to do that assignment just before the return i.e., change the use of s.Mounts on line 429 to also append to the local variable.

Reply:
Better yet -- I don't think we should be doing it at all @giggsoff ;-)